proposals/0000-proposal-process.rst

```diff
@@ -0,0 +1,319 @@
+   leads to repetitive and often difficult-to-follow discussions.
+
+3. The phases of the lifecycle of a proposal are not clearly defined.
+
+   This leads to unfortunate cases like that of ``ShortImports``, where
+   discussion proceeds for too long without a concrete specification. This
+   drains effort from both the author of the proposal and those discussing it.
+
+4. The process that exists isn't always consistently applied.
+
+   While this arguably has improved in recent years, there are cases where
+   features have been added without a formal specification or perhaps less
+   discussion than would be desired.
+
+
+Proposed change
+---------------
+
+I propose that we adopt a variant on the
+`Rust RFC scheme <https://github.com/rust-lang/rfcs#what-the-process-is>`_,
+building our proposal process on top of existing code review tools for
+dissemination and refinement of proposals with a standing committee acting as
+the final arbiter of proposal acceptance.
+
+To ensure that the process is as accessible and convenient as possible, I
+propose that we build this process on top of GitHub.
+
+Proposal process
+~~~~~~~~~~~~~~~~
+
+Each proposal goes through the following stages:
+
+1. The process begins with the opening of a pull request to the GHC proposals
+   repository, proposing the merge of a document (derived from a
+   `provided template <https://github.com/ghc-proposals/ghc-proposals/blob/master/0000-template.rst>`_)
+   specifying the proposed change. This document is the primary specification
+   of the feature and takes the place of the current Trac Wiki document.
+   
+2. Community members (including members of the proposal committee) will discuss
+   the proposal. The submitter is responsible for amending the specification to
+   account for collected comments. It should be expected that the proposal will
+   change during this process, which may last from days to months.
+
+3. When the author feels that the proposal is complete, and not being improved
+   by further discussion, they can submit the proposal to the proposal committee
+   for a decision.
+
+4. Within two (exceptionally three) weeks the committee will arrive at a
+   consensus as to whether the proposal in its submitted form meets a set of
+   criteria for inclusion (see below) while weighing the opinions expressed by
+   the community.
+
+   If the proposal is rejected then it will be sent back to the submitter along
+   with a rationale (referencing the criteria defined below). This is by no
+   means a final decision; the submitter may at any time amend and re-submit the
+   proposal to address the concerns of the committee.
+
+4. When the proposal is accepted the pull request will be merged and the
+   document will be preserved in the proposals repository as a permanent
+   specification for the feature.
+   
+5. The author will create a Trac ticket linking to the proposal to
+   provide a place track the progress of the implementation.
```

Comment:
This sounds perfectly reasonable to me. Moreover, it would be possible (with some effort) to track some portion of this metadata in the proposal itself.